From 52d353fbfff31a5456bbbe154b595761078363cf Mon Sep 17 00:00:00 2001
From: Jordan R Abrahams-Whitehead <ajordanr@google.com>
Date: Mon, 9 Mar 2026 19:22:17 +0000
Subject: [PATCH] Revert "v3: Make _LIBCPP_MAKE_OVERRIDABLE_FUNCTION_DETECTABLE
 a no-op on baremetal"

This reverts commit ff61f444b4c23a8a38e891cbdff0b393f7f7d203.

patch.cherry: false
patch.metadata.author: Jordan Abrahams-Whitehead <ajordanr@google.com>
patch.platforms: chromiumos
patch.version_range.from: 584947
patch.version_range.until: 584949

---
 libcxx/src/include/overridable_function.h | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/libcxx/src/include/overridable_function.h b/libcxx/src/include/overridable_function.h
index 2ca9eeb0227f..0b43f271486c 100644
--- a/libcxx/src/include/overridable_function.h
+++ b/libcxx/src/include/overridable_function.h
@@ -63,14 +63,8 @@
 // want to be defining special sections inside user's executables which use our headers.
 //
 
-// Baremetal targets tend to use custom linker scripts, which may not include
-// the __lcxx_override section.
-#if defined(LIBCXXABI_BAREMETAL)
+#if defined(_LIBCPP_OBJECT_FORMAT_MACHO)
 
-#  define _LIBCPP_CAN_DETECT_OVERRIDDEN_FUNCTION 0
-#  define _LIBCPP_MAKE_OVERRIDABLE_FUNCTION_DETECTABLE /* nothing */
-
-#elif defined(_LIBCPP_OBJECT_FORMAT_MACHO)
 #  define _LIBCPP_CAN_DETECT_OVERRIDDEN_FUNCTION 1
 #  define _LIBCPP_OVERRIDABLE_FUNCTION(type, name, arglist)                                                            \
     __attribute__((__section__("__TEXT,__lcxx_override,regular,pure_instructions"))) _LIBCPP_WEAK type name arglist
-- 
2.39.5